indexer-reader/src/test/java/org/apache/maven/index/reader/TestSupport.java

```diff
@@ -54,9 +56,7 @@
     public void setup()
         throws IOException
     {
-        this.tempDir = new File( "target/tmp-" + getClass().getSimpleName() );
-        this.tempDir.delete();
-        this.tempDir.mkdirs();
+        this.tempDir = Files.createTempDirectory( getClass().getSimpleName() + ".temp" ).toFile();
         this.directoryResourceHandlers = new ArrayList<>();
```

Comment:
I don't like creating temp directories in tests:
  * it's not deleted
  * it's hard to find
  * if the directory is deleted at the end of the test, its not available anymore to debug the test if needed

Reply:
We usually when needed to create temp dirs, use property for surefilre/failsafe to set tmp dir on CI to some place that is nuked along with workspace....

Reply:
I see https://github.com/apache/maven-indexer/blob/master/indexer-core/pom.xml#L179 which looks fine.